go-controller/pkg/ovn/ovn.go

```diff
@@ -877,7 +877,13 @@ func (oc *Controller) WatchNamespaces() {
 }
 
 func (oc *Controller) syncNodeGateway(node *kapi.Node, hostSubnets []*net.IPNet) error {
-	l3GatewayConfig, err := util.ParseNodeL3GatewayAnnotation(node)
+	var l3GatewayConfig, hybridGatewayConfig *util.L3GatewayConfig
+	var err error
+	if config.Gateway.Mode == config.GatewayModeHybrid {
```

Comment:
Can't do this. Check whether the node posts the hybridgateway annotation or not, and then you know whether it has hybrid enabled or not.

Reply:
Why not just adjust ParseNodeL3GatewayAnnotation() to return both structs, or an error?

Reply:
as we discussed we dont support mixing and matching gateway modes across nodes right now. In the future we probably should and can address this then.

Reply:
We need to pass this by Girish at least though. If we don't support this, it makes a number of things simpler. But if we do support this, then we should really support it everywhere. And that's definitely possible to do, fairly easily, in the code. We have the node name everywhere, though we may need to cache node annotations somewhere (like egressIP code).

Reply:
we talked about caching nodes in general didn't we? I remember that came up a few weeks ago. We could just store the gateway mode type there.